src/xcode/ENA/ENA/Source/Developer Menu/Features/DMWarnOthersNotificationViewController.swift

```diff
@@ -155,7 +178,23 @@ final class DMWarnOthersNotificationViewController: UIViewController, UITextFiel
 
 	private let store: Store
 	private var warnOthersReminder: WarnOthersRemindable
-
+	
+	private var currentSubmissionConsentStatusTitleLabel = UILabel()
+	
+	@objc
+	private func consentStateChanged(switchState: UISwitch) {
+		exposureSubmissionService.setSubmissionConsentGiven(consentGiven: switchState.isOn)
+		updateConsentStatusLabel()
+	}
+	
+	private func updateConsentStatusLabel() {
```

Comment:
Can't this be removed and handled with the sink in the init?

Reply:
Need to think on it for a moment 😉

Reply:
The function is also used in `viewDidLoad` for initial rendering.

Reply:
Completely refactored. Way more "combine" now.